### ISSUE TITLE
Support spatie/laravel-schemaless-attributes for json fields

### ISSUE THREAD

**Issue by ziming** [Laravel-Backpack/CRUD#1622]
Currently in backpack json fields need to be cast as array or object to work with some field types or columns

However many of us uses spatie/laravel-schemaless-attributes where we will get SchemalessAttributes when doing $model->json_field due to the getter method:

```
public function getExtraAttributesAttribute(): SchemalessAttributes
{
   return SchemalessAttributes::createForModel($this, 'extra_attributes');
}
```

Where we can still access attributes by $model->json_field['field1'] or $model->json_field->field1

Since spatie packages are hugely popular in the Laravel community. I think it will be good if support for it can be considered as well. So don't just check for is_array() and is_object(). Check for if it is SchemalessAttributes type too for determining if it is a json field

https://github.com/spatie/laravel-schemaless-attributes

Thank you

**Comment by tabacitu** [comment]
Hi @ziming ,

Thanks for the suggestion - sounds like we can add schemaless-attributes support pretty easily, and it might be a good idea - but the idea itself doesn’t look very popular - nobody commented so far. Also, I see this Spatie package isn’t very popular either, it only has 22k downloads - which is not a lot by their standards.

Let’s see if this proposal gathers any more support - we’ll provide SchemalessAttributes support if more people ask for it. Until then, I’m going to close this - I know people who search for this will definitely find it - your issue is well-written and has all the keywords it needs. Thanks again!

Out of curiosity - where did Backpack stumble when working with schemaless attributes? How did you work around it?

Cheers!

**TLDR: If you want support for schemaless-attributes please reply to this issue and say so, and we’ll reopen the issue and consider it.**

**Comment by ziming** [comment]
I created the json editor field pull request earlier. You can check it out

**Comment by ziming** [comment]
I only comment on that to show you where it stumble and how my field handles that

**Comment by tabacitu** [comment]
Yup - as I expected, such an easy fix :-) Thank you!

**Comment by vesper8** [comment]
I would like support for json fields to be added as part of the core
